### PR TITLE
URA-872 - tests for resuming partially uploaded runs

### PR DESCRIPTION
**Pull request by jethror1**
- end to end tests for testing the behaviour of a run partially uploaded, then resuming on the next call to ensure interrupted uploads continue as expected
- new functions added to `helper.py`
  - these are simple functions with reused behaviour for all tests abstracted here
- fix typos in `__init__.py`

```
jethro@jethro-T490:~/Projects/s3_upload/tests/e2e$ python3 -m pytest -v .
============================================================== test session starts ===============================================================
platform linux -- Python 3.8.10, pytest-7.0.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.8.10', 'Platform': 'Linux-5.15.0-124-generic-x86_64-with-glibc2.29', 'Packages': {'pytest': '7.0.1', 'pluggy': '1.0.0'}, 'Plugins': {'Faker': '14.0.0', 'repeat': '0.9.3', 'subtests': '0.11.0', 'html': '4.1.0', 'metadata': '3.0.0', 'mock': '3.11.1', 'cov': '4.0.0', 'random-order': '1.1.1'}}
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
rootdir: /home/jethro/Projects/s3_upload/tests/e2e
plugins: Faker-14.0.0, repeat-0.9.3, subtests-0.11.0, html-4.1.0, metadata-3.0.0, mock-3.11.1, cov-4.0.0, random-order-1.1.1
collected 23 items                                                                                                                               

test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_no_errors_in_logs PASSED                                   [  4%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_remote_files_upload_correctly PASSED                       [  8%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_slack_post_message_after_uploading_as_expected PASSED      [ 13%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_upload_log_file_for_fully_uploaded_run_correct PASSED      [ 17%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_uploaded_run_not_picked_up_to_upload_again PASSED          [ 21%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_when_nothing_to_upload_that_exit_code_is_zero PASSED       [ 26%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_no_errors_in_logs PASSED                     [ 30%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_no_upload_log_generated_for_run_3 PASSED     [ 34%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_remote_files_uploaded_correctly PASSED       [ 39%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_slack_post_message_after_uploading_as_expected PASSED [ 43%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_upload_log_files_for_fully_uploaded_run_1_correct PASSED [ 47%]
test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_upload_log_files_for_fully_uploaded_run_2_correct PASSED [ 52%]
test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_complete_upload_log_file_as_expected PASSED                         [ 56%]
test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_error_message_for_failed_upload_in_partial_upload_log PASSED        [ 60%]
test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_partially_uploaded_log_file_as_expected PASSED                      [ 65%]
test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_slack_post_messages_as_expected PASSED                              [ 69%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_no_errors_in_logs PASSED       [ 73%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_remote_file_uploaded_correctly PASSED [ 78%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_slack_post_message_after_uploading_as_expected PASSED [ 82%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_upload_log_files_for_fully_uploaded_run_1_correct PASSED [ 86%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_upload_log_files_for_fully_uploaded_run_2_correct PASSED [ 91%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_uploaded_run_not_picked_up_to_upload_again PASSED [ 95%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_when_nothing_to_upload_that_exit_code_is_zero PASSED [100%]

=============================================================== 23 passed in 6.00s ===============================================================
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/31)

### CODE REVIEW

File: tests/e2e/helper.py [hunk 1]
```diff
@@ -21,3 +32,73 @@ def create_files(run_dir, *files):
...
+    """
+    Read in the run upload log file
+
+    Returns
+    -------
+    dict
+        contents of run upload log file
+    """
+    with open(
+        path.join(TEST_DATA_DIR, "logs/uploads/run_1.upload.log.json"),
```
Comment (fletchercr) [inline tests/e2e/helper.py:83]: wouldn't the run name for this change and not always be `run_1` ?

File: tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py [hunk 2]
```diff
@@ -107,29 +109,8 @@ def setUpClass(cls):
...
-        for log_file in glob(os.path.join(TEST_DATA_DIR, "logs", "*log*")):
-            os.remove(log_file)
-
-        # clean up the remote files we just uploaded
-        bucket = boto3.resource("s3").Bucket(S3_BUCKET)
-        objects = bucket.objects.filter(Prefix=cls.parent_remote_path)
-        bucket.delete_objects(
-            Delete={"Objects": [{"Key": obj.key} for obj in objects]}
-        )
+        cleanup_local_test_files(cls.run_1, cls.run_2)
```
Comment (fletchercr) [inline tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py:114]: do we need to call `cleanup_remote_files` here too?

File: tests/e2e/helper.py (hunk 1 above)
Comment (fletchercr) [reply tests/e2e/helper.py:83]: ah, I see this is only called for the testing of the partial runs, so you wouldn't need to generalise it for multiple log.json files

**Review by jethror1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/31)* status: 0 of 10 files reviewed, 2 unresolved discussions (waiting on @fletchercr)

___
*[`tests/e2e/helper.py` line 83 at r2](https://reviewable.io/reviews/eastgenomics/s3_upload/31#gh-1827832571:-OArs-_X73yQgxfsk3IG:b-896fix) ([raw file](https://github.com/eastgenomics/s3_upload/blob/a302c395d9b255cb3f680e1fe3e8678f5b7ab233/tests/e2e/helper.py#L83)):*
<details><summary><i>Previously, fletchercr wrote…</i></summary><blockquote>

ah, I see this is only called for the testing of the partial runs, so you wouldn't need to generalise it for multiple log.json files
</blockquote></details>

Done.
___
*[`tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py` line 114 at r2](https://reviewable.io/reviews/eastgenomics/s3_upload/31#gh-1827845255:-OArs3Tk1vPbyvgigddB:b45dtb) ([raw file](https://github.com/eastgenomics/s3_upload/blob/a302c395d9b255cb3f680e1fe3e8678f5b7ab233/tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py#L114)):*
<details><summary><i>Previously, fletchercr wrote…</i></summary><blockquote>

do we need to call `cleanup_remote_files` here too?
</blockquote></details>

yep good spot, fixed it

**Review by fletchercr — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 2 of 2 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/31)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @jethror1)

**Pull request merged**